careers/software-engineer.md

```diff
@@ -57,3 +101,17 @@ An engineer III is capable of taking on complex technical problems and coordinat
 **Communication:** They point out novel opportunities for refactoring during code review. They proactively work to minimize or mitigate potential problems before they arise.
 
 **Collaboration:** Others on the team reach out for help in their area of expertise, and they make themselves available. They are capable of effectively running small to medium-sized projects with little to no coaching, but need help to successfully manage large projects.
+
+### Responsibilities
+
+* Lead the design and development of medium to large-scale software features or systems.
+* Establish and contribute to team-level software architecture and development standards.
+* Mentor and provide technical guidance to junior and mid-level engineers.
+* Collaborate with cross-functional teams to define requirements and contribute to technical roadmaps.
+* Drive the adoption of best practices around coding, design, quality, and performance within your team.
+* Troubleshoot and debug complex technical issues, often involving multiple systems.
+* Participate in code reviews, ensuring that the code we deliver is of the highest possible quality.
```

Comment:
I like the "They point out novel opportunities for refactoring during code review." bit from the "Communication" section above -- could we try including that here?

> Improve team code quality through insightful reviews that identify non-obvious refactoring opportunities and architectural improvements.

Not attached to that specific language, but the idea that an Engineer III is finding overlooked and insightful opportunities to improve the code is important.